test-data/unit/stubgen.test

```diff
@@ -552,14 +602,148 @@ def f(a): ...
 [case testInferOptionalOnlyFunc]
 class A:
     x = None
-    def __init__(self, a=None) -> None:
+    def method(self, a=None):
         self.x = []
 [out]
 from typing import Any, Optional
 
 class A:
-    x = ...  # type: Any
-    def __init__(self, a: Optional[Any] = ...) -> None: ...
+    x: Any
+    def method(self, a: Optional[Any] = ...): ...
+
+[case testAnnotationImportsFrom]
+import foo
+from collection import defaultdict
+x: defaultdict
+
+[out]
+from collection import defaultdict
+
+x: defaultdict
+
+[case testAnnotationImports]
+import foo
+import collection
+x: collection.defaultdict
+
+[out]
+import collection
+
+x: collection.defaultdict
+
+
+[case testAnnotationImports]
+from typing import List
+import collection
+x: List[collection.defaultdict]
+
+[out]
+import collection
+from typing import List
+
+x: List[collection.defaultdict]
+
+
+[case testAnnotationFwRefs]
+x: C
+
+class C:
+    attr: C
+
+y: C
+[out]
+x: 'C'
```

Comment:
I didn't know that (and a lot of the patch could be made simpler by removing forward ref handling). I couldn't find that documented anywhere (looking at PEP 484); is that actually a defined behaviour or an implementation artifact?